### ISSUE TITLE
Optimzer should be stepped before scheduler

### ISSUE THREAD

**Issue by sarthakpati** [CBICA/GaNDLF#107]
**Describe the bug**
Currently, GaNDLF calls `optimizer.step()` **after** `scheduler.step()` but it should be the other way around. n PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-

**Additional context**
Reported by @Geeks-Sid

**Comment by sarthakpati** [comment]
GaNDLF is calling `optimizer.step()` in the `training_loop.train_network` function [[ref](https://github.com/CBICA/GaNDLF/blob/master/GANDLF/training_loop.py#L197)] and `scheduler.step()` in the `training_loop.validate_network` function [[ref](https://github.com/CBICA/GaNDLF/blob/master/GANDLF/training_loop.py#L607)], which is exactly what is expected.
